include/jemalloc/internal/tsd_types.h

```diff
@@ -291,7 +291,8 @@ a_name##tsd_cleanup_wrapper(void) {					\
 	if (wrapper == NULL) {						\
 		return false;						\
 	}								\
-	if (a_cleanup != malloc_tsd_no_cleanup &&			\
+	if (/* a_cleanup != malloc_tsd_no_cleanup && */			\
+	    /* Verify compiler (mingw32+gcc) bug.*/			\
```

Comment:
Confirmed compiler bug. On Windows with MinGW64, the "a_cleanup != malloc_tsd_no_cleanup" is not computed correctly and is compiled to false incorrectly, even though we never actually passed in malloc_tsd_no_cleanup. This resulted in missing cleanups. Without this line of change (and L404), the test failure is reproducible in Windows, e.g. https://ci.appveyor.com/project/jasone/jemalloc/build/638/job/ulyou1yumv1b1o6a

I'll update with a workaround.